evap/results/views.py

```diff
@@ -124,14 +127,16 @@ def course_detail(request, semester_id, course_id):
                     if show_grades:
                         contributor_sections[section.contributor]['total_votes'] += result.total_count
 
-    course.avg_grade, course.distribution = calculate_average_grade_and_distribution(course)
+    if show_grades:
+        course.avg_grade, course.distribution = calculate_average_grade_and_distribution(course)
+    else:
+        course.avg_grade, course.distribution = None, None
 
     template_data = dict(
             course=course,
             course_sections_top=course_sections_top,
             course_sections_bottom=course_sections_bottom,
             contributor_sections=contributor_sections,
-            show_grades=show_grades,
```

Comment:
in this file, is that a functional change? if yes, could you explain what's the change?

Reply:
i saw that, the question was whether the templates now render something different since they don't get this boolean but instead None grades.

Reply:
no, just the gray bar. before they also didn't show the grade and deviation.

Reply:
except for the first two hunks the changes in this file are missing their template changes, which seem to be in the next commit. ideally, they go into a separate commit along with their template changes, because the next commit is already a functional change and no refactoring.